packages/python-packages/doc-warden/warden/HeaderConstruct.py

```diff
@@ -0,0 +1,38 @@
+import re
+
+class HeaderConstruct():
+    def __init__(self, tag, parent, patterns = []):
+        self.tag = tag
+        self.parent = parent
+        self.level = 0
+        self.matching_patterns = []
+
+        if tag:
+            self.tag = tag
+            self.level = int(tag.name.replace('h', ''))
+
+        for pattern in patterns[:]:
+            if re.search(pattern, self.tag.get_text()):
+                self.matching_patterns.append(pattern)
+
+    def __str__(self):
+      return self.tag.get_text()
+
+    def get_tag_text(self):
+        return str(self)
+
+    def check_parents_for_pattern(self, pattern):
+        current_parent = self.parent
+
+        while current_parent:
+            result = re.search(pattern, current_parent.get_tag_text())
+
+            if result:
+                return current_parent
+            else:
+                if current_parent.parent:
+                    current_parent = current_parent.parent
+                else:
+                    return None
```

Comment:
just a nit. I think you can just replace this with `current_parent = current_parent.parent` and it will still work the same way if parent's parent is None

Reply:
👍